### PR TITLE
ETFE-1441 Build new page for ED01 "delay-type"

### PR DESCRIPTION
**Pull request by grahampaulcook**
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,44 @@
 
-# emcs-tfe-explain-delay-frontend
+# emcs-tfe-report-a-receipt-frontend
```
Comment (james-work-account) [inline README.md]:
```suggestion
# emcs-tfe-explain-delay-frontend
```

File: app/config/Module.scala [hunk 2]
```diff
@@ -17,7 +17,7 @@
 package config
 
 import com.google.inject.AbstractModule
-import controllers.actions.{AuthAction, AuthActionImpl, DataRetrievalAction, DataRetrievalActionImpl, MovementAction, MovementActionImpl}
+import controllers.actions.{AuthAction, AuthActionImpl, DataRequiredAction, DataRequiredActionImpl, DataRetrievalAction, DataRetrievalActionImpl, MovementAction, MovementActionImpl, UserAllowListAction, UserAllowListActionImpl}
```
Comment (james-work-account) [inline app/config/Module.scala]:
```suggestion
import controllers.actions._
```
Might as well just import it all at this point (that's what IntelliJ suggests anyway)

File: conf/messages.cy [hunk 3]
```diff
@@ -69,3 +69,17 @@ notOnPrivateBeta.notSignedUp.p1.preLink = You can choose to
 notOnPrivateBeta.notSignedUp.p1.link = take part in the EMCS private beta research
 notOnPrivateBeta.notSignedUp.p1.afterLink = if you are not already involved.
 notOnPrivateBeta.notSignedUp.p2 = To take part you must currently submit EMCS receipts using the HMRC platform.
+
+arc.subHeading = Explanation for delay to {0}
```
Comment (james-work-account) [inline conf/messages.cy:73]: Copydeck says "Explain a delay to [ARC]" but prototype says "Explanation for delay to [ARC]" - worth checking which is correct with UCD guys.

File: conf/messages.en [hunk 4]
```diff
@@ -69,3 +69,17 @@ notOnPrivateBeta.notSignedUp.p1.preLink = You can choose to
 notOnPrivateBeta.notSignedUp.p1.link = take part in the EMCS private beta research
 notOnPrivateBeta.notSignedUp.p1.afterLink = if you are not already involved.
 notOnPrivateBeta.notSignedUp.p2 = To take part you must currently submit EMCS receipts using the HMRC platform.
+
+arc.subHeading = Explanation for delay to {0}
```
Comment (james-work-account) [inline conf/messages.en:73]: See comment on messages.cy

File: app/views/DelayTypeView.scala.html [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+            @govukButton(
+                ButtonViewModel(messages("site.continue"))
+            )
+        }
+    }
+
+    @{
+        //$COVERAGE-OFF$
+    }
```
Comment (james-work-account) [inline app/views/DelayTypeView.scala.html]:
```suggestion
@layout(pageTitle = title(form, messages("delayType.title"))) {

    @formHelper(action = routes.DelayTypeController.onSubmit(request.ern, request.arc, mode), 'autoComplete -> "off") {

        @if(form.errors.nonEmpty) {
            @govukErrorSummary(ErrorSummaryViewModel(form, errorLinkOverrides = Map("value" -> "value_0")))
        }

        @h1("delayType.heading", Some(messages("arc.subHeading", request.arc)), "govuk-heading-l")

        @p(classes = "govuk-body") {
            @messages("delayType.p1")
        }

        @p(classes = "govuk-body") {
            @messages("delayType.p2")
        }

        @govukRadios(
            RadiosViewModel(
                field = form("value"),
                legend = LegendViewModel(HtmlContent(messages("delayType.title"))).withCssClass("govuk-fieldset__legend--m"),
                items = DelayType.options
            )
        )

        @govukButton(
            ButtonViewModel(messages("site.continue"))
        )
    }
}

@{
    //$COVERAGE-OFF$
}
```

Indentation is a bit weird here - just needs moving back 1

**Review by james-work-account — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (grahampaulcook) [reply README.md]: Updated

File: app/config/Module.scala (hunk 2 above)
Comment (grahampaulcook) [reply app/config/Module.scala]: Updated

File: app/views/DelayTypeView.scala.html (hunk 5 above)
Comment (grahampaulcook) [reply app/views/DelayTypeView.scala.html]: Updated

File: conf/messages.cy (hunk 3 above)
Comment (grahampaulcook) [reply conf/messages.cy:73]: Asked UCD, see reply https://hmrcdigital.slack.com/archives/C03NB3T2D42/p1688734280541539?thread_ts=1688734226.950419&cid=C03NB3T2D42

File: conf/messages.en (hunk 4 above)
Comment (grahampaulcook) [reply conf/messages.en:73]: Asked UCD, see reply https://hmrcdigital.slack.com/archives/C03NB3T2D42/p1688734280541539?thread_ts=1688734226.950419&cid=C03NB3T2D42

**Review by james-work-account — APPROVED** [review]

**Pull request merged**